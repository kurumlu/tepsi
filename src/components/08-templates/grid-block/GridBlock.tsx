import * as React from 'react';
import { FunctionComponent } from 'react';
import cn from 'classnames';

export enum GridBlockVariant {
  // Single item
  BLOCK = '',
  ROW = 'row',
  COLUMN = 'col',

  // Two items
  BLOCKS = 'blocks',
  ROWS = 'rows',
  COLUMNS = 'cols',

  // Three items
  ROW_PLUS_COLUMNS = 'row-cols',
  ROWS_PLUS_COLUMN = 'rows-col',
  COLUMN_PLUS_ROWS = 'col-rows',
  COLUMNS_PLUS_ROW = 'cols-row',
}

export type GridBlockProps = {
  variant?: GridBlockVariant;
};

export const GridBlock: FunctionComponent<GridBlockProps> = ({
  variant = GridBlockVariant.BLOCK,
}) => {
  const getGridItemAmount = () => {
    switch (variant) {
      case GridBlockVariant.BLOCK:
      case GridBlockVariant.ROW:
      case GridBlockVariant.COLUMN:
        return 1;
      case GridBlockVariant.BLOCKS:
      case GridBlockVariant.ROWS:
      case GridBlockVariant.COLUMNS:
        return 2;
      default:
        return 3;
    }
  };
  const gridItemAmount = getGridItemAmount();

  return (
    <div
      className={cn('sd-t-grid-block', {
        [`sd-t-grid-block--${variant}`]: variant,
      })}>
      <div className="sd-t-grid-block__inner">
        {[...Array(gridItemAmount)].map((_, index) => (
          <div className="sd-t-grid-block__item" key={index}>
            <div className="sd-t-grid-block__item-content-placeholder"></div>
          </div>
        ))}
      </div>
    </div>
  );
};
