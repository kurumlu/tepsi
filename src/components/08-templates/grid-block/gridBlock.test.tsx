import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { GridBlock, GridBlockProps, GridBlockVariant } from './GridBlock';

describe('<GridBlock />', () => {
  it('renders default GridBlock component', () => {
    const component: ShallowWrapper<GridBlockProps> = shallow(<GridBlock />);
    expect(component.find('.sd-t-grid-block__item')).toHaveLength(1);
    expect(component).toMatchSnapshot('default');
  });

  it('renders row GridBlock component', () => {
    const component: ShallowWrapper<GridBlockProps> = shallow(
      <GridBlock variant={GridBlockVariant.ROW} />
    );
    expect(component.find('.sd-t-grid-block__item')).toHaveLength(1);
    expect(component).toMatchSnapshot('row');
  });

  it('renders column GridBlock component', () => {
    const component: ShallowWrapper<GridBlockProps> = shallow(
      <GridBlock variant={GridBlockVariant.COLUMN} />
    );
    expect(component.find('.sd-t-grid-block__item')).toHaveLength(1);
    expect(component).toMatchSnapshot('column');
  });

  it('renders rows GridBlock component', () => {
    const component: ShallowWrapper<GridBlockProps> = shallow(
      <GridBlock variant={GridBlockVariant.ROWS} />
    );
    expect(component.find('.sd-t-grid-block__item')).toHaveLength(2);
    expect(component).toMatchSnapshot('rows');
  });

  it('renders columns GridBlock component', () => {
    const component: ShallowWrapper<GridBlockProps> = shallow(
      <GridBlock variant={GridBlockVariant.COLUMNS} />
    );
    expect(component.find('.sd-t-grid-block__item')).toHaveLength(2);
    expect(component).toMatchSnapshot('columns');
  });

  it('renders blocks GridBlock component', () => {
    const component: ShallowWrapper<GridBlockProps> = shallow(
      <GridBlock variant={GridBlockVariant.BLOCKS} />
    );
    expect(component.find('.sd-t-grid-block__item')).toHaveLength(2);
    expect(component).toMatchSnapshot('blocks');
  });

  it('renders row + columns GridBlock component', () => {
    const component: ShallowWrapper<GridBlockProps> = shallow(
      <GridBlock variant={GridBlockVariant.ROW_PLUS_COLUMNS} />
    );
    expect(component.find('.sd-t-grid-block__item')).toHaveLength(3);
    expect(component).toMatchSnapshot('row + columns');
  });

  it('renders rows + column GridBlock component', () => {
    const component: ShallowWrapper<GridBlockProps> = shallow(
      <GridBlock variant={GridBlockVariant.ROWS_PLUS_COLUMN} />
    );
    expect(component.find('.sd-t-grid-block__item')).toHaveLength(3);
    expect(component).toMatchSnapshot('rows + column');
  });

  it('renders column + rows GridBlock component', () => {
    const component: ShallowWrapper<GridBlockProps> = shallow(
      <GridBlock variant={GridBlockVariant.COLUMN_PLUS_ROWS} />
    );
    expect(component.find('.sd-t-grid-block__item')).toHaveLength(3);
    expect(component).toMatchSnapshot('column + rows');
  });

  it('renders columns + row GridBlock component', () => {
    const component: ShallowWrapper<GridBlockProps> = shallow(
      <GridBlock variant={GridBlockVariant.COLUMNS_PLUS_ROW} />
    );
    expect(component.find('.sd-t-grid-block__item')).toHaveLength(3);
    expect(component).toMatchSnapshot('columns + row');
  });
});
