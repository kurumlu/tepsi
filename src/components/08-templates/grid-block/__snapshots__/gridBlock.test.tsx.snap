// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridBlock /> renders blocks GridBlock component: blocks 1`] = `
<div
  className="sd-t-grid-block sd-t-grid-block--blocks"
>
  <div
    className="sd-t-grid-block__inner"
  >
    <div
      className="sd-t-grid-block__item"
      key="0"
    >
      <div
        className="sd-t-grid-block__item-content-placeholder"
      />
    </div>
    <div
      className="sd-t-grid-block__item"
      key="1"
    >
      <div
        className="sd-t-grid-block__item-content-placeholder"
      />
    </div>
  </div>
</div>
`;

exports[`<GridBlock /> renders column + rows GridBlock component: column + rows 1`] = `
<div
  className="sd-t-grid-block sd-t-grid-block--col-rows"
>
  <div
    className="sd-t-grid-block__inner"
  >
    <div
      className="sd-t-grid-block__item"
      key="0"
    >
      <div
        className="sd-t-grid-block__item-content-placeholder"
      />
    </div>
    <div
      className="sd-t-grid-block__item"
      key="1"
    >
      <div
        className="sd-t-grid-block__item-content-placeholder"
      />
    </div>
    <div
      className="sd-t-grid-block__item"
      key="2"
    >
      <div
        className="sd-t-grid-block__item-content-placeholder"
      />
    </div>
  </div>
</div>
`;

exports[`<GridBlock /> renders column GridBlock component: column 1`] = `
<div
  className="sd-t-grid-block sd-t-grid-block--col"
>
  <div
    className="sd-t-grid-block__inner"
  >
    <div
      className="sd-t-grid-block__item"
      key="0"
    >
      <div
        className="sd-t-grid-block__item-content-placeholder"
      />
    </div>
  </div>
</div>
`;

exports[`<GridBlock /> renders columns + row GridBlock component: columns + row 1`] = `
<div
  className="sd-t-grid-block sd-t-grid-block--cols-row"
>
  <div
    className="sd-t-grid-block__inner"
  >
    <div
      className="sd-t-grid-block__item"
      key="0"
    >
      <div
        className="sd-t-grid-block__item-content-placeholder"
      />
    </div>
    <div
      className="sd-t-grid-block__item"
      key="1"
    >
      <div
        className="sd-t-grid-block__item-content-placeholder"
      />
    </div>
    <div
      className="sd-t-grid-block__item"
      key="2"
    >
      <div
        className="sd-t-grid-block__item-content-placeholder"
      />
    </div>
  </div>
</div>
`;

exports[`<GridBlock /> renders columns GridBlock component: columns 1`] = `
<div
  className="sd-t-grid-block sd-t-grid-block--cols"
>
  <div
    className="sd-t-grid-block__inner"
  >
    <div
      className="sd-t-grid-block__item"
      key="0"
    >
      <div
        className="sd-t-grid-block__item-content-placeholder"
      />
    </div>
    <div
      className="sd-t-grid-block__item"
      key="1"
    >
      <div
        className="sd-t-grid-block__item-content-placeholder"
      />
    </div>
  </div>
</div>
`;

exports[`<GridBlock /> renders default GridBlock component: default 1`] = `
<div
  className="sd-t-grid-block"
>
  <div
    className="sd-t-grid-block__inner"
  >
    <div
      className="sd-t-grid-block__item"
      key="0"
    >
      <div
        className="sd-t-grid-block__item-content-placeholder"
      />
    </div>
  </div>
</div>
`;

exports[`<GridBlock /> renders row + columns GridBlock component: row + columns 1`] = `
<div
  className="sd-t-grid-block sd-t-grid-block--row-cols"
>
  <div
    className="sd-t-grid-block__inner"
  >
    <div
      className="sd-t-grid-block__item"
      key="0"
    >
      <div
        className="sd-t-grid-block__item-content-placeholder"
      />
    </div>
    <div
      className="sd-t-grid-block__item"
      key="1"
    >
      <div
        className="sd-t-grid-block__item-content-placeholder"
      />
    </div>
    <div
      className="sd-t-grid-block__item"
      key="2"
    >
      <div
        className="sd-t-grid-block__item-content-placeholder"
      />
    </div>
  </div>
</div>
`;

exports[`<GridBlock /> renders row GridBlock component: row 1`] = `
<div
  className="sd-t-grid-block sd-t-grid-block--row"
>
  <div
    className="sd-t-grid-block__inner"
  >
    <div
      className="sd-t-grid-block__item"
      key="0"
    >
      <div
        className="sd-t-grid-block__item-content-placeholder"
      />
    </div>
  </div>
</div>
`;

exports[`<GridBlock /> renders rows + column GridBlock component: rows + column 1`] = `
<div
  className="sd-t-grid-block sd-t-grid-block--rows-col"
>
  <div
    className="sd-t-grid-block__inner"
  >
    <div
      className="sd-t-grid-block__item"
      key="0"
    >
      <div
        className="sd-t-grid-block__item-content-placeholder"
      />
    </div>
    <div
      className="sd-t-grid-block__item"
      key="1"
    >
      <div
        className="sd-t-grid-block__item-content-placeholder"
      />
    </div>
    <div
      className="sd-t-grid-block__item"
      key="2"
    >
      <div
        className="sd-t-grid-block__item-content-placeholder"
      />
    </div>
  </div>
</div>
`;

exports[`<GridBlock /> renders rows GridBlock component: rows 1`] = `
<div
  className="sd-t-grid-block sd-t-grid-block--rows"
>
  <div
    className="sd-t-grid-block__inner"
  >
    <div
      className="sd-t-grid-block__item"
      key="0"
    >
      <div
        className="sd-t-grid-block__item-content-placeholder"
      />
    </div>
    <div
      className="sd-t-grid-block__item"
      key="1"
    >
      <div
        className="sd-t-grid-block__item-content-placeholder"
      />
    </div>
  </div>
</div>
`;
